rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2975,6 +2976,27 @@ public String getLoginFromSessionId(@PathParam("sessionId") String sessionId) {
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    @GET
+    @Path("logins/sessionid/{sessionId}/userdata")
+    @Produces("application/json")
+    public UserData getUserDataFromSessionId(@PathParam("sessionId") String sessionId) {
+        if (sessionId != null && sessionStore.exists(sessionId)) {
+            try {
+                Scheduler scheduler = sessionStore.get(sessionId).getScheduler();
+                UserData userData = scheduler.getCurrentUserData();
+                return userData;
+            } catch (NotConnectedException e) {
+                return null;
```

Comment:
There is currently no authorization mechanism for the rest API directly, and here we simply check if a session exists. And if it exists, we call the getCurrentUserData() method which has no specific security policy (i.e. an authenticated user can always know his username or his group). 
I don't know well how to check access here.